quantuminspire/util/configuration.py

```diff
@@ -210,6 +210,13 @@ async def _wait_until_token_becomes_valid(access_token: str) -> None:
         token_issued_at = int(decoded_token["iat"])
 
         while True:
-            await asyncio.sleep(1)
-            if int(time.time()) > token_issued_at:
+            current_time = int(time.time())
+            time_diff = current_time - token_issued_at
+
+            if time_diff > 0:
                 return
+            
+            await asyncio.sleep(abs(time_diff))
```

Comment:
Why do you subtrackt the iat from current_time? You have to do an `abs` in that case. Also, if time_diff is 0, you don't have to wait either. You only have to wait if the time difference is strictly larger than 0. 

```suggestion
            time_diff = token_issued_at - current_time

            if time_diff > 0:
                await asyncio.sleep(time_diff)
            return
```

Reply:
fixed